hasUniqueChars.c

```diff
@@ -77,6 +80,55 @@ bool hasUniqueChars(char * inputStr) {
   // It also illustrates how to use the seeBits function for debugging.
   // Printed values should initially be all zeros
   // TODO: remove or comment out this code when satisfied of function correctness
+
+  for (int i = 0; i < strlen(inputStr); i++) {
+    nextChar = inputStr[i];
+    unsigned long index = nextChar-65;
+    unsigned long mask = 1l << index;
+    /// This is saying that if there is a space to continue
+    if (nextChar == 32) {
+      continue;
+    }
+    /// This is checking if nextChar is between A-Z and if so then to get an index and mask to determine whether the nextChar is repeated in the input string
+    else if ((nextChar >= 65) && nextChar <=90) {
+      unsigned long index = nextChar - 65;
+      unsigned long mask = 1l << index;
+      /// Checking to see if there is a match
+      if (mask & checkBitsA_z) {
+        return false;
+      }
+      /// Tracks to see if it has been seen and if not to save it as seen return false if there is a duplicate
+      else {
+        checkBitsA_z |= mask;
+      }
+    }
+    /// This is checking if nextChar is between a-z and if so to get an index and mask to determine whether the nextChar is repeated in the input string
+    else if ((nextChar >= 97) && nextChar <=122) {
+      unsigned long index = nextChar - 97 + 26;
```

Comment:
this should all be part of the same range as the previous one, don't leave out the non-alpha characters in this range!